buildtest/utils/file.py

```diff
@@ -143,3 +143,66 @@ def resolve_path(path):
 
     if os.path.exists(real_path):
         return real_path
+
+
+def read_file(filepath):
+    """ This method provides capability to read a file.
+
+        Parameters:
+
+        :param filepath: file name to read
+        :type filepath: str, required
+        :return: return content of file
+        :rtype: list
+    """
+
+    # type check filepath to ensure its a string, if not return None
+    if not isinstance(filepath, str):
+        return None
+
+    # resolve_path will handle shell and user expansion and account for any symlinks and check for file existence.
+    # if resolve_path does not return gracefully it implies file does not exist and will return None
+    filepath = resolve_path(filepath)
+
+    # if it's invalid file let's return None
+    if not filepath:
+        return None
+
+    with open(filepath, "r") as fd:
+        content = fd.read()
+
+    return content
+
+
+def write_file(filepath, content):
+    """ This method provides capability to write a file.
+
+        Parameters:
+
+        :param filepath: file name to write
+        :type filepath: str, required
+        :param content: content to write to file
+        :type content: str, required
+        :return: return content of file
+        :rtype: list
+    """
+
+    # type check filepath to ensure its a string, if not return None
+    if not isinstance(filepath, str):
+        return None
+
+    # ensure content is of type string
+    if not isinstance(content, str):
+        return None
+
+    # if filepath is an actual file, let's not write to file and return None.
```

Comment:
This isn't expected behavior - I would want to give a write file function a file (that already might exist) and have it over-written. If you are concerned about this, then there could be a check in the calling function, but it shouldn't be here with the util (and then silently not update the file that exists, that sounds like a bad silent bug to me!)

Reply:
Okay i removed this check so you can overwrite to file.  This is added in https://github.com/shahzebsiddiqui/buildtest-framework/commit/e4ad6f2188b00e085e92d94b225b8587e1c03cdf#diff-b94bc4259405fb65c0cd50bf50dc3bc1L200